src/codec.rs

```diff
@@ -29,11 +29,18 @@ pub mod pb_codec {
     use protobuf::{CodedInputStream, Message};
 
     use super::MessageReader;
-    use crate::error::Result;
+    use crate::error::{Error, Result};
 
     #[inline]
-    pub fn ser<T: Message>(t: &T, buf: &mut Vec<u8>) {
-        t.write_to_vec(buf).unwrap()
+    pub fn ser<T: Message>(t: &T, buf: &mut Vec<u8>) -> Result<()> {
+        t.write_to_vec(buf)?;
+        if buf.len() <= u32::MAX as usize {
```

Comment:
Add a comment to explain that the source of this number is from https://github.com/grpc/grpc/blob/master/doc/PROTOCOL-HTTP2.md